File: public/data/chapters/chapter_01_arithmetic/logic.js
// Logic for Addition in JavaScript
// Expects 'context' to be available in scope
// Must define 'result' variable

const a = context.a || 0;
const b = context.b || 0;
const sum = a + b;

// Log evidence (simulated by returning a structured object, 
// though the runtime only captures the final result)
// In a real scenario, we might want to emit logs via a side channel or return them.
// For this demo, we just return the result.

var result = sum;

File: public/data/chapters/chapter_01_arithmetic/logic.py
"""
Arithmetic Logic for Python Runtime.
Context is injected into the global scope by the runtime.
"""

def add(target_content):
    """
    Compute sum of 'a' and 'b' from context.
    """
    # 'context' is available in the global scope
    a = context.get("a", 0)
    b = context.get("b", 0)
    return a + b

def multiply(target_content):
    """
    Compute product of 'a' and 'b' from context.
    """
    a = context.get("a", 1)
    b = context.get("b", 1)
    return a * b
